components/SwipeListView.js

```diff
@@ -25,6 +25,12 @@ class SwipeListView extends PureComponent {
                 onContentSizeChange: (w, h) => this.onContentSizeChange(w, h),
             };
         }
+        this._onScroll = this.onScroll.bind(this);
+        if (typeof this.props.onScroll === 'object') {
```

Comment:
Can you explain what's going on here? When would `onScroll` be an object?

Reply:
Hi @jemise111, I couldn't find any benchmark on performance comparison between the two either, but I found this in the source code. Here, Animated.FlatList is FlatList wrapped in another component (which is returned from createAnimatedComponent function and handles native events). I don't have concrete data to back this up but it looks like by default if you opts in to use animation with native driver without specifying scroll event throttle, you might even get better performance due to events being emitted less frequently through the bridge.
```
const FlatList = require('../../../Lists/FlatList');
const createAnimatedComponent = require('../createAnimatedComponent');

import type {AnimatedComponentType} from '../createAnimatedComponent';

/**
 * @see https://github.com/facebook/react-native/commit/b8c8562
 */
const FlatListWithEventThrottle = React.forwardRef((props, ref) => (
  <FlatList scrollEventThrottle={0.0001} {...props} ref={ref} />
));

module.exports = (createAnimatedComponent(
  FlatListWithEventThrottle,
): AnimatedComponentType<
  React.ElementConfig<typeof FlatList>,
  React.ElementRef<typeof FlatList>,
>);
```

When AnimatedEvent object created with Animated.event([], { useNativeDrive }) is passed to onScroll, it will received an object instead of a callback function as also mentioned in #331 
```
const event = function(
  argMapping: $ReadOnlyArray<?Mapping>,
  config: EventConfig,
): any {
  const animatedEvent = new AnimatedEvent(argMapping, config);
  if (animatedEvent.__isNative) {
    return animatedEvent;
  } else {
    return animatedEvent.__getHandler();
  }
};
```

Reply:
@ocarlott didn't think I would get the answer from my own comment 🤦‍♂ 😂 

Thanks for the additional context, this looks good, going to merge and release it now!

(If on the off chance I see any user complaints about performance I may reach out asking (if you have time) if you can help me diagnose the issue)

Reply:
Sure thing, I would love to help anyone has any issue with this PR.
I'm putting some examples here in case anyone may find this helpful:

JS thread event only:
```
const jsScroll = new Animated.Value(0);
<SwipeListView
  onScroll={(event) => {
    jsScroll.setValue(event.nativeEvent.contentOffset.y);
  }}
/>
```

Native driver event only:
```
const nativeScroll = new Animated.Value(0);
<SwipeListView
  onScroll={Animated.event([
    { 
      nativeEvent: {
        contentOffset: {
          y: nativeScroll
        }
      }
    }], 
   { useNativeDriver: true }
  )}
/>
```

Native driver event and js thread event:

```
const nativeScroll = new Animated.Value(0);
const jsScroll = new Animated.Value(0);

<SwipeListView
  onScroll={Animated.event([
    { 
      nativeEvent: {
        contentOffset: {
          y: nativeScroll
        }
      }
    }], 
   { 
     useNativeDriver: true,
     listener: (event) => {
       jsScroll.setValue(event.nativeEvent.contentOffset.y);
     }
   }
  )}
/>
```